apps/designer/app/canvas/custom-components/link.tsx

```diff
@@ -0,0 +1,48 @@
+import { useLocation } from "@remix-run/react";
+import type {
+  ElementRef,
+  ComponentProps,
+  ForwardRefExoticComponent,
+  RefAttributes,
+} from "react";
+import { forwardRef } from "react";
+import { preserveSearchBuildParams } from "~/shared/router-utils";
+
+// href starts with http:, mailto:, tel:, etc.
+const isAbsoluteUrl = (href: string) => /^[a-z]+:/i.test(href);
```

Comment:
inetresting that `new URL('hello:world');` works and defines hello as protocol. Always thought that '//' is needed.
probably safe way is 
```js
try {
  new URL(href); 
  return true
}
catch { return false }
```
but have no strong opinion